src/tools/generate/lazy_load_images.js

```diff
@@ -0,0 +1,9 @@
+const lazy_load_images = (html) => {
+  const re = /<img\s/gi;
```

Comment:
My only concern is that this is so broad that it'll pick up things like:

```html
<pre>
  <img href="example.jpg">
</pre>
```

where the image HTML is part of the actual content.

One way to mitigate this problem is to limit it to `<img` within a `<figure>` tag.

Reply:
I know what you mean, I was also worried about that. It doesn't seem to have picked any up, yet (I did a spot check on the generated content).

I think it would be better to explicitly exclude `<img>` inside a `<pre>` tag rather than only look for them inside a `<figure>` tag. I know it would be possible with regex, but that may be beyond my skill level to do. If we try and do it via JSDOM, we will run into the same serialization issues that plagued the first attempt at `generate_figure_id`.

Reply:
But if we do only look for images inside `<figure>`, do you think that [this regex would suffice](https://regex101.com/r/F0zR4w/2)?

Reply:
I much prefer this - means the chapter images and other logos will not be lazy loaded.

Just tried that regex with 3 figure images and it seems to only pick up the last one. Will have a play with it and see what I can come up with.

Reply:
This one seems to be better: https://regex101.com/r/RAGFDE/1

Reply:
Oh cool! Thanks! I'll incorporate that one.

Reply:
Added a test case for tables to prove they don't get picked up: https://regex101.com/r/RAGFDE/3

Reply:
Hm. The only problem with being more specific, is that not every chapter has put the images inside figures. I honestly believe that it may be better to be overly enthusiastic about putting `loading="lazy"` everywhere, than missing out most of the `<img>` tags - but I stand to be corrected.

I'm committing both regexes and we can swap between them.

Reply:
Every chapter should have images in figures and will have soon as part of copy editing. And if they don't then they just miss out on lazy loading - so not worst thing in the world.

Not sure on impact of being overly enthusiastic. Will the main chapter image, plus any logos of the like (which should not be lazy loaded) load slightly later as the browser takes a bit longer to figure out whether they are needed or not? Micro seconds extra maybe, but I still like trying to do it right.

Reply:
Plus, we forgot about the main reason we did this more specific call was to address @rviscomi's concern:

> My only concern is that this is so broad that it'll pick up things like:
> 
> ```
> <pre>
>   <img href="example.jpg">
> </pre>
> ```
> 
> where the image HTML is part of the actual content.
> 
> One way to mitigate this problem is to limit it to `<img` within a `<figure>` tag.

Reply:
I think this is working as expected, now.